src/sagemaker_training/process.py

```diff
@@ -201,6 +209,12 @@ def check_error(cmd, error_class, processes_per_host, cwd=None, capture_error=Tr
         extra_info = None
         if return_code == 137:
             extra_info = "OutOfMemory: Process killed by SIGKILL (signal 9)"
+        error_class = errors.ExecuteUserScriptError
+        for error_name in error_classes:
+            if error_name in stderr:
```

Comment:
Will there be a scenario where there be multiple errors reported. Example: NCCL related error eventually causing mpi to fail (MPI related error) ? If so, are we settling for the first error that is seen in the `error_classes` ?

Reply:
we are only interested in specific error that will attribute a Deep Engine library caused this error. So we will not scan for general MPI ABORT or NCCL error. Error names in error_classes as all targeted library specific errors and usually only 1 type of error in log. Even if multiple are found, only 1 is enough for error attribution.(Blame the library for causing the error). So we are good here.